presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSplit.java

```diff
@@ -93,15 +99,28 @@ public FileFormat getFileFormat()
         return partitionKeys;
     }
 
+    @JsonProperty
     @Override
     public NodeSelectionStrategy getNodeSelectionStrategy()
     {
-        return NO_PREFERENCE;
+        return nodeSelectionStrategy;
     }
-
     @Override
     public List<HostAddress> getPreferredNodes(List<HostAddress> sortedCandidates)
     {
+        if (sortedCandidates == null || sortedCandidates.isEmpty()) {
+            throw new PrestoException(NO_NODES_AVAILABLE, "sortedCandidates is null or empty for HiveSplit");
+        }
+
+        if (getNodeSelectionStrategy() == SOFT_AFFINITY) {
+            // Use + 1 as secondary hash for now, would always get a different position from the first hash.
+            int size = sortedCandidates.size();
+            int mod = path.hashCode() % size;
+            int position = mod < 0 ? mod + size : mod;
+            return ImmutableList.of(
+                sortedCandidates.get(position),
+                sortedCandidates.get((position + 1) % size));
+        }
         return addresses;
```

Comment:
I guess these lines are copied from HiveSplit, can we extract these lines into a util method?  then we can share the implementation.   But I'm fine if you or someone else would like to do it in another PR, your call.

Reply:
Yeah, it copied from HiveSplit, i try to abstract a common method to place this logic to a util class, but I cannot decide whether we could add a new dependency `guava` to the `presto-common`, `presto-spi` or some other module. As I'm not familiar enough with presto now, it would be nice to do this method abstraction in another PR, thank you.